### PR TITLE
V3: Refactors codap modal 

### PR DESCRIPTION
**Pull request by eireland**
The refactor moves the render of the modal content to the child instead of the modal level

### CODE REVIEW

File: v3/src/index.scss [hunk 1]
```diff
@@ -11,12 +11,12 @@ body {
...
-      // chakra uses box-shadow for focus highlight rather than outline
-      box-shadow: var(--chakra-shadows-outline) !important;
-    }
-  }
+  // .chakra-button {
+  //   &:focus-visible {
+  //     // chakra uses box-shadow for focus highlight rather than outline
+  //     box-shadow: var(--chakra-shadows-outline) !important;
+  //   }
+  // }
```
Comment (kswenson) [inline v3/src/index.scss:19]: Is this change related to this PR? This was done to prevent distracting highlighting when clicking while not disabling highlighting for keyboard focus. Do you have reason to believe it's no longer necessary?

**Review by kswenson — CHANGES_REQUESTED** [review]
👍 Dialog changes look good -- some of the resulting dialog code is quite lengthy and so we might want to look into ways to break it up a bit, but that need not be part of this PR. The one question/suggestion is about the commenting out of the `focus-visible` rule.

File: v3/src/index.scss (hunk 1 above)
Comment (eireland) [reply v3/src/index.scss:19]: Highlighting wasn't coming up on keyboard focus. But you're right, it shouldn't be part of this PR

**Review by kswenson — APPROVED** [review]
👍 LGTM
